### PR TITLE
CB-13455 Enabling CMEK for GCP to encrypt FREEIPA disk

### PR DESCRIPTION
**Pull request by cegganesh84**
This change enables GCP CMEK for for resource encryption. In this change we have introduced a new parameter encryption-key which accept the GCP key resource path and encrypts the FREEIPA disk using this encryption key.
See the design doc for more details. https://docs.google.com/document/d/199zU-ECjfk4B0nIBzYq_5G6T7NM1uCCuxuVYG7_fwws/edit#heading=h.lqrcqpih6dy

Changes include:

Adding new param encryption_key to gcp-create-environment
GCP CMEK Key validation using regex and storing encryption-key in enviroment-parameters table.
Using GCP CMEK key to encrypt FREEIPA Disk by setting keyEncryptionMethod to KMS. If KeyEncryotionMethod is set to KMS then GcpDiskResourceBuilder verifies this and uses key present in VOLUME_ENCRYPTION_KEY_ID using CustomGcpDiskEncryptionService to encrypt the vm disk.
Sample Payload : https://drive.google.com/file/d/1alm7pT4J71oTZILxGHePPDTVO3w9mho4/view

Testing

Relevant unit tests have been added.
Tested locally using postman by creating environment using encryption key and without encryption key. The behaviour of disk allocation is as expected.

See detailed description in the commit message.

### CODE REVIEW

**Comment by cegganesh84** [comment]
It looks like there is a bad merge here, which I did not catch locally using `./gradlew build`. Doing a clean build first.

File: freeipa/src/test/java/com/sequenceiq/freeipa/converter/stack/CreateFreeIpaRequestToStackConverterTest.java [hunk 1]
```diff
@@ -119,34 +119,22 @@ void testConvertForInstanceGroupsWhendiskEncryptionSetIdIsPresent() {
...
         underTest.convert(source, environmentResponse, ACCOUNT_ID, owner, "crn1", CloudPlatform.AZURE.name());
-        verify(instanceGroupConverter).convert(
-                any(InstanceGroupRequest.class),
-                any(NetworkRequest.class),
-                anyString(),
-                any(Stack.class),
-                any(FreeIpaServerRequest.class),
-                any(DetailedEnvironmentResponse.class),
-                anyString());
+        assertEquals(mapCaptorForEncryption.getValue().get(DISK_ENCRYPTION_SET_ID), "dummyDiskEncryptionSetId");
```
Comment (cegganesh84) [inline freeipa/src/test/java/com/sequenceiq/freeipa/converter/stack/CreateFreeIpaRequestToStackConverterTest.java:137]: This test case is failing. I am not sure what broke. I see that this assertion was removed during multi-AZ implementation. https://github.com/hortonworks/cloudbreak/pull/11215#discussion_r699708326

CC: @doktoric @juhi-09 @lajosrodek

Comment (cegganesh84) [reply freeipa/src/test/java/com/sequenceiq/freeipa/converter/stack/CreateFreeIpaRequestToStackConverterTest.java]: Strangely removing this mocking fixes the issue, `when(networkConverter.convert(any(NetworkRequest.class))).thenReturn(new Network());`

**Comment by cegganesh84** [comment]
@lajosrodek @doktoric @juhi-09 Could you please kindly review. I am trying to unblock QE in mow-dev.

**Review by doktoric — APPROVED** [review]

**Pull request merged**